nextjs-13/src/app/page.tsx

```diff
@@ -31,6 +37,23 @@ const Home = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  async function openCreateModal() {
+    setModalType('create');
+    setIsOpen(true);
+  }
+
+  async function openUpdateModal(item) {
```

Comment:
Why isn't this defined in the `ListNewsletter` component? It seems out of scope of the root page 🤔

Reply:
Hmm I see.

The reason I do that is that, although it's true defining `openUpdateModal` is not the responsibility of the root page, but without defining `openUpdateModal` here, then instead of we only need to pass one prop, which is the `openUpdateModal` to `ListNewsletter`, we would need to instead pass [3 props into the `ListNewsletter`, which is the `setModalType`, `setIsOpen`, and `setCurrentNewsletter`](https://github.com/nimblehq/js-framework-benchmark/pull/118/files#diff-223e9b714cf2c9b5f163079f0bd63d3a6eac9c46915fc3afde20a2742897d706R46-R55) so that the `ListNewsletter` can create it's own `openUpdateModal`.

So in short, I did that just to make it looks more concise.

What do you think about this? Is passing 3 props instead of 1 is the better practice we prefer taking? 👀

Reply:
Understood. In this simple case, having the parent owning the state is consistent with [the official best practice](https://react.dev/learn/sharing-state-between-components#a-single-source-of-truth-for-each-state). In more complex applications or use cases, it would be best to use `Context` and `useReducer` (see https://stackoverflow.com/a/63687665)